### PR TITLE
Added solution to spr-03

### PR DESCRIPTION
**Pull request by MichaelShevchenko**

### CODE REVIEW

File: src/main/java/com/spring/practice/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    @GetMapping
+    public List<UserResponseDto> getAll() {
+        List<UserResponseDto> result = new ArrayList<>();
+        for (User user : userService.listUsers()) {
+            result.add(castUserToDto(user));
+        }
+        return result;
+    }
+
+    private UserResponseDto castUserToDto(User user) {
```
Comment (Nokafan) [inline src/main/java/com/spring/practice/controller/UserController.java:47]: Don’t pass User in the constructor of UserResponseDto, let’s use setters.

**Review by Nokafan — COMMENTED** [review]
Good job

**Review by dariiabutorina — APPROVED** [review]

File: src/main/java/com/spring/practice/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    @GetMapping("/{userId}")
+    public UserResponseDto get(@PathVariable Long userId) {
+        return castUserToDto(userService.getById(userId));
+    }
+
+    @GetMapping
+    public List<UserResponseDto> getAll() {
+        List<UserResponseDto> result = new ArrayList<>();
```
Comment (RomanDubovskyi) [inline src/main/java/com/spring/practice/controller/UserController.java]: checklist 
https://mate-academy.github.io/jv-program-fulltime/04_spring/checklist/hw_32.html

Comment (MichaelShevchenko) [reply src/main/java/com/spring/practice/controller/UserController.java]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**